packages/@aws-cdk/aws-lambda-nodejs/lib/bundling.ts

```diff
@@ -2,136 +2,208 @@ import * as lambda from '@aws-cdk/aws-lambda';
 import * as cdk from '@aws-cdk/core';
 import * as fs from 'fs';
 import * as path from 'path';
-import { findPkgPath } from './util';
+import { PackageJsonManager } from './package-json-manager';
+import { findClosestPathContaining } from './util';
 
 /**
- * Options for Parcel bundling
+ * Base options for Parcel bundling
  */
-export interface ParcelOptions {
+export interface ParcelBaseOptions {
   /**
-   * Entry file
+   * Whether to minify files when bundling.
+   *
+   * @default false
    */
-  readonly entry: string;
+  readonly minify?: boolean;
 
   /**
-   * Expose modules as UMD under this name
+   * Whether to include source maps when bundling.
+   *
+   * @default false
    */
-  readonly global: string;
+  readonly sourceMaps?: boolean;
 
   /**
-   * Minify
+   * The cache directory
+   *
+   * Parcel uses a filesystem cache for fast rebuilds.
+   *
+   * @default - `.cache` in the root directory
    */
-  readonly minify?: boolean;
+  readonly cacheDir?: string;
 
   /**
-   * Include source maps
+   * The root of the project. This will be used as the source for the volume
+   * mounted in the Docker container. If you specify this prop, ensure that
+   * this path includes `entry` and any module/dependencies used by your
+   * function otherwise bundling will not be possible.
+   *
+   * @default - the closest path containing a .git folder
    */
-  readonly sourceMaps?: boolean;
+  readonly projectRoot?: string;
 
   /**
-   * The cache directory
+   * The environment variables to pass to the container running Parcel.
+   *
+   * @default - no environment variables are passed to the container
    */
-  readonly cacheDir?: string;
+  readonly containerEnvironment?: { [key: string]: string; };
```

Comment:
The fact that you are running parcel inside a container is an implementation detail from users' perspective, perhaps `parcelEnvironment` or just `env`

Reply:
went for `parcelEnvironment` because we already have `environment` from the Lambda props.

Reply:
Cool